Add documentation for the RuntimeCleaner and Crawdad daemonset

Brian-McM opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][DOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

Product Version(s):
<!--- Specify which versions of Calico, Calico Enterprise, and Calico Cloud your PR applies to. -->

Issue:
<!--- Add a link to the Jira ticket or GitHub issue, if applicable. --->

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

SME review:
- [ ] An SME has approved this change. 
<!--- SME approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

DOCS review:
- [ ] A member of the docs team has approved this change.
<!-- The Docs team must review and approve all changes. -->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from docs team:
  For community authors: tag @tigera/docs to ask for a review when your PR is ready --->

stevegaossou commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,21 @@
+When Runtime View is enabled in Scan Settings, {{prodname}} deploys a Daemonset, named "Crawdad", into your connected Managed Clusters to
+actively monitor pods running in your Kubernetes clusters and gather image information from the node
+they're running on. This information is used to correlate the images your pods are running with the images you
+have upload scan results for.
```
Is this supposed to be: 

> have **uploaded** scan results for.

Or: 

> have scan results for.

stevegaossou commented:
@Brian-McM I noticed the preview version of this PR does not include your changes. It would be worth checking with the docs team to see whether preview works for all PRs or not. 

I think there's some extra configuration you need to at least have the new page included in the side menu bar that's currently missing in this PR.

ghost commented:
I am going through some test cycles for this feature, and the "Enable Runtime View" toggle was a bit confusing to me. I added a new cluster to my management cluster and I expected to have to enable the feature for this specific cluster, but it was already enabled. Can we change the language to show the customer that the toggle for this enables it across all their clusters? Also, is the score thresholds universal as well? Some language to explain this would be great as well.

Brian-McM commented:
@stevegaossou there's tweaks we can make to the Makefile target to show this, but currently cloud is pinned to show only 3.15 changes (which is not where this belongs). They haven't made the 3.16 section yet, but I'll check to see if there's any precedent on setting this for the PR so it's viewable.

ctauchen commented:
@Brian-McM Some context would be helpful. Can you add some information so the docs team know what we're reviewing? Thanks!

Brian-McM commented:
Need to add information about setting the CR socket path, for rancher (or anything using cri-dockerd) it's `/var/run/cri-dockerd.sock`, for kubernetes versions before 1.23 (or anything else using dockershim) it's `/var/run/dockershim.sock`

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,21 @@
+
+# Configuring the Runtime View
+
+**Enable Runtime View**
```
Need to show / describe how to enable the run time view, not just what happens when you do.

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,27 @@
+
+:::warning
+
+The Runtime View requires that all the connected Managed Clusters are running Kubernetes using a CRI for container management.
```
Maybe state kubernetes versions and link to something explaining what the CRI is.

Brian-McM committed:
Add documentation for enabling / configuring the runtime view

This PR separates out the details of the runtime view into it's own document, and does a more comprehensive overview of what it does, how you enable it, and how it needs to be configured in special cases.

It outlines pre requisites around requiring k8s clusters that use CRI, and highlights what components are installed and what privileges they need.

Brian-McM committed:
Remove CRI requirement section (CRI has been around since K8s 1.5)

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+
+## How does the Runtime View collect image information from my Managed Clusters
+
+When Runtime View is enabled, {{prodname}} deploys a Daemonset, named "Crawdad", into your connected Managed Clusters to
```
No, this is intentional. It's how the grammar works for product names.

Brian-McM committed:
Don't move all runtime view info to new page, just make a page describing how to change the cri socket path

ctauchen reviewed (commented):
@Brian-McM This structure is much more friendly to our users. Ping me when you've got the doc itself ready to review.

Brian-McM committed:
Add files describing how to change the socket path

Brian-McM committed:
Add note about root privileges

stevegaossou commented on calico-cloud/image-assurance/change-the-cri-socket-path.mdx:
````diff
@@ -0,0 +1,41 @@
+kubectl patch imageassurance default --type='merge' -p '{"spec":{"criSocketPath":"<path-to-cri-socket"}}'
+```
+
+See [the installation reference](../reference/installation/api.mdx#image-assurance.operator.tigera.io/v1.ImageAssuranceSpec) for more details on the ImageAssurance CR.
````
@Brian-McM should we call out somewhere that if the daemon set is not deployed correctly, the user can check the tigerastatus for more information?

Brian-McM replied:
I think we'll need to do a follow up PR to re work the runtime view docs. @ctauchen had some ideas about making the overall description of the runtime view better.

Phil had said nobody is using this currently, so we can skip the initial release notes and warnings, so for now we're just going to highlight how to configure it to work if the default configuration doesn't work.

Brian-McM commented on calico-cloud/image-assurance/change-the-cri-socket-path.mdx:
```diff
@@ -0,0 +1,41 @@
+
+## Big picture
+
+Change the CRI socket path in connected Managed Clusters from the path
```
That was supposed to be "from the default path", thanks for the catch.

Brian-McM committed:
Update grammar issues

Brian-McM commented on calico-cloud/image-assurance/scan-image-registries.mdx:
```diff
@@ -222,6 +222,8 @@ In the Scan Results and Runtime tabs, the **Registry path** field may be blank i
 
 :::
 
+For information configuring the Runtime View please see [Configuring the Runtime View](configuring-the-runtime-view.mdx)
```
Not applicable anymore, restructure the docs.

Brian-McM commented on calico-cloud/reference/installation/api.mdx:
```diff
@@ -5,7 +5,14 @@ description: Installation API reference
+
+## Image Assurance installation reference
+
+The Kuberenetes resources below configure Image Assurance installation when using the operator.
```
changes installation to installations instead

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+
+## Value
+
+Runtime View collects image information from all the pods running in all of your connected Managed Clusters. This information is linked to uploaded image scan results so you can see if you clusters are running vulnerable images.
```
Change the structure, N/A

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+might be located other than the default location. If the Crawdad Pods running in your Managed Cluster don't have the correct
+location of the CRI socket, they won't be able to gather image information for the pods running in your cluster.
+
+Other than containerd, we know of two other CRI implementations (there may be others):
```
Change the structure, N/A

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+
+Ensure that all Managed Clusters connected to your Management Cluster are running a Kubernetes using a CRI for container management.
+
+**Enable Runtime View**
```
Change the structure, N/A

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+**Configuring the CRI socket path for Managed Cluster**
+
+Some Kubernetes clusters use different CRI binaries other than the default containerd one, or the containerd CRI socket
+might be located other than the default location. If the Crawdad Pods running in your Managed Cluster don't have the correct
```
Change the structure, N/A

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+To enable the Runtime View go to the Scan Settings and check the "Enable Runtime View" box. The settings can be
+accessed by clicking the gear on the right side of the page in the Scan Results tab.
+
+By default, this feature is disabled. If you disable the feature after it has been enabled, runtime information that was
```
No it doesn't remove it.

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+
+## Value
+
+Runtime View collects image information from all the pods running in all of your connected Managed Clusters. This information is linked to uploaded image scan results so you can see if you clusters are running vulnerable images.
```
Change the structure, N/A

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+
+## Big picture
+
+View pods running vulnerability images for all your Managed Clusters.
```
Change the structure, N/A

Brian-McM commented on calico-cloud/image-assurance/configuring-the-runtime-view.mdx:
```diff
@@ -0,0 +1,63 @@
+---
+description: Use the Runtime View to find vulnerabilities in your cluster.
```
Change the structure, N/A

Brian-McM committed:
Fix more grammtical errors

ctauchen commented:
Changes tracked now in https://github.com/tigera/docs/pull/581
